feat(mock-server): add proxy

alex-held opened the pull request.

alex-held committed:
feat(mock-server): add mock-server api

alex-held committed:
feat(mock-server): match request

* matches path
* matches method

alex-held committed:
feat(mock-server): match request by header

ghost commented:
### Features

*  **mock-server:** add mock-server api ([7449418](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/74494187d9056fc578f8d1044a065045a3ab3566))
*  **mock-server:** match request ([9e57466](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/9e57466baba46d886792848a536b8253284f3bef))
*  **mock-server:** match request by header ([da7a1db](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/da7a1db7b434326af2ad45f4ba66083c6fa43baf))
*  **mock-server:** match request by string body ([9b36a5c](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/9b36a5c3f53c66b977228c36b432df2e7e2f8878))
*  **mock-server:** match request by query string ([f1743aa](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/f1743aa2d4f45079fbb46c6514fd563d3cd25071))
*  **mock-server:** add docker support ([4089188](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/4089188f867d1084b2c3b0f217c0b8a65ceeef67))


### Styles

* apply ReSharper code-styless ([264d1bd](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/264d1bd781b107dc802b1812e52dc3bfd840902b))


### Code Refactoring

*  **mock-server:** add test for SetupService :white_check_mark: ([9547c55](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/9547c550366a823f5a0d17a8c6dc3ba7a609a89b))
*  **mock-server:** wrap ISetupRepository in ISetupService ([d5cbd84](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/d5cbd848305c1c3eb8a3556679e4fb58a22f7e8b))
*  **mock-server:** annotate non/-nullable reference types ([306fe10](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/306fe10176e845d69650e96e449d4393396857f5))
*  **mock-server:** create setups via SetupController ([2387bde](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/2387bdeae0f1d5fdbd68a3f4d773a8d4762a17d5))
*  **mock-server:** use ports 80 (matching) + 1080 (admin) ([592dec1](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/592dec13897250dba2f6c2974852647e3749bbd6))


### Styles

* apply ReSharper code styles ([538f3ad](https://github.com/alex-held/NinjaTools.FluentMockServer/commit/538f3ad3475897ef77008ec2889e8e226dace817))


### Contributors

@alex-held

alex-held committed:
Merge branch 'develop' into feature/mock-server

alex-held committed:
feat(mock-server): match request by string body

* exact string body
* partial string body

alex-held committed:
style: apply ReSharper code-styless

alex-held committed:
refactor(mock-server): add test for SetupService :white_check_mark:

alex-held committed:
feat(mock-server): match request by query string

alex-held committed:
refactor(mock-server): wrap ISetupRepository in ISetupService

alex-held committed:
!refactor(mock-server): determine admin area from port

!BREAKING CHANGE: the admin area used to be calculated via the 'mock-server' path segment

alex-held committed:
feat(mock-server): add docker support

alex-held committed:
styles: apply ReSharper code styles

alex-held committed:
refactor(mock-server): annotate non/-nullable reference types

alex-held committed:
refactor(mock-server): create setups via SetupController

* disable implicit [Required] model binding handling

alex-held committed:
refactor(mock-server): use ports 80 (matching) + 1080 (admin)
